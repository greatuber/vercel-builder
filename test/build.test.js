const path = require('path')
const runBuildLambda = require('./utils/run-build-lambda')

const FOUR_MINUTES = 240000

it('Should build the standard example', async () => {
  const { buildResult, cacheResult } = await runBuildLambda(
    path.join(__dirname, 'fixture')
  )

  // Lambda
  expect(buildResult.index).toBeDefined()

  // Build files
  const buildFiles = [
    'test.txt',
    '_nuxt/LICENSES'
  ]
  for (const file of buildFiles) {
    expect(buildResult[file]).toBeDefined()
  }

  // Cache files
  const cacheFiles = [
    'node_modules_dev/@babel/core/package.json',
    'node_modules_prod/@nuxt/core-edge/package.json'
  ]
  for (const file of cacheFiles) {
    expect(cacheResult[file]).toBeDefined()
  }
}, FOUR_MINUTES)
