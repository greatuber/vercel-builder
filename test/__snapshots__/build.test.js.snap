// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should build the standard example 1`] = `
Array [
  "index",
  "_nuxt/08001e3704db5d6f0c4a.js",
  "_nuxt/259f4864a00b6ff19cd3.js",
  "_nuxt/f3c502d6c98ebea29c1a.js",
  "_nuxt/LICENSES",
]
`;
